### PR TITLE
Dockerize

### PR DESCRIPTION
**Pull request by dustin-hawkins**
* Adds a `Dockerfile` that replaces the /build scripts
* Adds a docker-compose file with some sane defaults, and sample configuration files
* Updates the README.md with quickstart instructions for using docker-compose

### CODE REVIEW

**Comment by dustin-hawkins** [comment]
closes #4

**Comment by dustin-hawkins** [comment]
One note, if you removed the need for the conf.json, and instead allowed all of those variables to be passed in via the linux environment, it would be much more flexible in deploying to production environments - like Kubernetes.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM golang:alpine
+WORKDIR /src
+RUN apk add --update npm git
+RUN go get -u github.com/jteeuwen/go-bindata/...
+COPY ./webapp/package.json webapp/package.json
```
Comment (dustin-hawkins) [inline Dockerfile]: By copying only package.json, then doing the NPM install - we speed up subsequent builds where there the package.json does not change.

npm install is so slow.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+RUN cd ./webapp && \
+    npm install
+COPY . .
+RUN cd ./webapp && \
+    npm run build
```
Comment (dustin-hawkins) [inline Dockerfile]: same functionality as /build/build_webapp.sh

**Comment by amborle** [comment]
I am following the instructions, but I get the following:

```
docker-compose build
WARNING: The FEATMAP_DB_PASSWORD variable is not set. Defaulting to a blank string.
WARNING: The FEATMAP_DB_USER variable is not set. Defaulting to a blank string.
WARNING: The FEATMAP_DB variable is not set. Defaulting to a blank string.
WARNING: The FEATMAP_HTTP_PORT variable is not set. Defaulting to a blank string.
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.featmap.ports is invalid: Port ranges don't match in length
```

I guess the .env should be copied **before** running `docker-compose build`?

**Comment by amborle** [comment]
I copied the .env file and continue the instructions, and get the containers to run.  Unfortunately, I get this:
```
docker logs featmap_featmap_1
2019/08/28 11:11:04 database error
```

I have not changed something .env and conf.json.

**Comment by amborle** [comment]
I was able to get this out:
```
postgres_1  | 2019-08-29 06:10:33.111 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres_1  | 2019-08-29 06:10:33.111 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres_1  | 2019-08-29 06:10:33.113 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres_1  | 2019-08-29 06:10:33.437 UTC [1] LOG:  database system is ready to accept connections
postgres_1  | 2019-08-29 06:10:43.463 UTC [30] FATAL:  role "postgres" does not exist
postgres_1  | 2019-08-29 06:10:44.582 UTC [31] FATAL:  password authentication failed for user "featmap"
postgres_1  | 2019-08-29 06:10:44.582 UTC [31] DETAIL:  Password does not match for user "featmap".
postgres_1  |   Connection matched pg_hba.conf line 95: "host all all all md5"
featmap_1   | 2019/08/29 06:10:44 database error
```
So,  seems like featmap can connect to the database but runs in to some kind of authentication problem.

**Comment by dustin-hawkins** [comment]
Starting from a fresh copy of the repository, on my dockerize branch - I run the following and everything comes up:

------
```
cp config/env.sample .env
cp config/conf.sample.json config/conf.json
docker-compose up
```
The parameters in the `/.env` file need to match the params in the `config/conf.json` file, so if you changed the username or password, make sure to update the `dbConnectionString` value in the json.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                image: postgres:11-alpine
+                restart: always
+                volumes:
+                        - ${PWD}/data:/var/lib/postgresql/data
+                environment:
+                        POSTGRES_DB: ${FEATMAP_DB}
+                        POSTGRES_USER: ${FEATMAP_DB_USER}
+                        POSTGRES_PASSWORD: ${FEATMAP_DB_PASSWORD}
+                healthcheck:
+                        test: ["CMD-SHELL", "pg_isready -U postgres"]
```
Comment (amborle) [inline docker-compose.yml]: The user `postgres` does not exist, I think this needs to be changed to:

`test: ["CMD-SHELL", "pg_isready -U ${FEATMAP_DB_USER}"]`

**Pull request merged**